File: iOS/AUiKit/AUiKit/Classes/Model/AUiRoomConfig.swift
//
//  AUiRoomConfig.swift
//  AUiKit
//
//  Created by wushengtao on 2023/2/24.
//

import Foundation

open class AUiCommonConfig: NSObject {
    /// appid
    public var appId: String = ""
    /// 网络请求域名
    public var host: String = ""
    
    //用户信息
    public var userId: String = ""
    public var userName: String = ""
    public var userAvatar: String = ""
    
    open override var description: String {
        return "AUiCommonConfig: userId: \(userId) userName: \(userName)"
    }
}

open class AUiRoomConfig: NSObject {
    public var channelName: String = ""     //正常rtm使用的频道
    public var rtmToken007: String = ""     //rtm login用，只能007
    public var rtcToken007: String = ""     //rtm join用
    
    public var rtcChannelName: String = ""  //rtc使用的频道
    public var rtcRtcToken: String = ""  //rtc join使用
    public var rtcRtmToken: String = ""  //rtc mcc使用
    
    public var rtcChorusChannelName: String = ""  //rtc 合唱使用的频道
    public var rtcChorusRtcToken: String = ""  //rtc 合唱join使用
}

File: Android/auikit/src/main/java/io/agora/auikit/model/AUiRoomConfig.java
package io.agora.auikit.model;

import android.util.SparseArray;
import android.view.View;

import androidx.annotation.NonNull;

public class AUiRoomConfig {

    @NonNull public String channelName = "";     //正常rtm使用的频道
    @NonNull public String rtmToken007 = "";     //rtm login用，只能007
    @NonNull public String rtcToken007 = "";     //rtm join用

    @NonNull public String rtcChannelName = "";  //rtc使用的频道
    @NonNull public String rtcRtcToken006 = "";  //rtc join使用
    @NonNull public String rtcRtmToken006 = "";  //rtc mcc使用，只能006
    @NonNull public String rtcChorusChannelName = "";  //rtc 合唱使用的频道
    @NonNull public String rtcChorusRtcToken007 = "";  //rtc 合唱join使用

    public int themeId = View.NO_ID;

    public AUiRoomConfig(String roomId) {
        channelName = roomId;
        rtcChannelName = roomId + "_rtc";
        rtcChorusChannelName = roomId + "_rtc_ex";
    }
}
